KarhooUISDK/Routing/BookingFlowCoordinator.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  BookingFlowCoordinator.swift
+//  KarhooUISDK
+//
+//  Created by Diana Petrea on 04.04.2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+
+final class BookingFlowCoordinator: KarhooUISDKSceneCoordinator {
+    var baseViewController: BaseViewController
+    var navigationController: UINavigationController?
+    var childCoordinators: [KarhooUISDKSceneCoordinator] = []
+    
+    private var bookingCompletion: ((BookingScreenResult) -> Void)?
+    
+    static let shared = BookingFlowCoordinator()
```

Comment:
Why do we need singletone for this one?

Reply:
To have only one per session. I was also considering making the class public to allow the DP to start the booking flow from here, something like `BookingFlowCoordinator.shared.start()`